source/rock/middle/FunctionCall.ooc

```diff
@@ -1255,6 +1259,32 @@ FunctionCall: class extends Expression {
                                     return null
                                 }
                             }
+                        } else if (retType) {
+                            if (implArg instanceOf?(FunctionDecl)) {
```

Comment:
`instanceOf? + as` is a form I avoid in rock code nowadays.

This one is preferred:

```ooc
match implArg {
  case fDecl: FunctionDecl =>
    argType := fDecl inferredReturnType
    // etc.
}
```